cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/cruisecontrol/CruiseControlApiImpl.java

```diff
@@ -0,0 +1,239 @@
+                            JsonObject json = buffer.toJsonObject();
+                            CruiseControlRebalanceResponse ccResponse = new CruiseControlRebalanceResponse(userTaskID, json);
+                            result.complete(ccResponse);
+                        });
+                    } else if (response.statusCode() == 202) {
+                        response.bodyHandler(buffer -> {
+                            String userTaskID = response.getHeader(CC_REST_API_USER_ID_HEADER);
+                            JsonObject json = buffer.toJsonObject();
+                            CruiseControlRebalanceResponse ccResponse = new CruiseControlRebalanceResponse(userTaskID, json);
+                            if (json.containsKey(CC_REST_API_PROGRESS_KEY)) {
+                                // If the response contains a "progress" key then the rebalance proposal has not yet completed processing
+                                ccResponse.setProposalIsStillCalculating(true);
+                            } else {
+                                result.fail(new CruiseControlRestException("202 Status code did not contain progress key. Response was: " +
+                                        ccResponse.getJson().toString()));
+                            }
+                            result.complete(ccResponse);
+                        });
+                    } else if (response.statusCode() == 500) {
+                        response.bodyHandler(buffer -> {
+                            String userTaskID = response.getHeader(CC_REST_API_USER_ID_HEADER);
+                            JsonObject json = buffer.toJsonObject();
+                            CruiseControlRebalanceResponse ccResponse = new CruiseControlRebalanceResponse(userTaskID, json);
+                            if (json.containsKey(CC_REST_API_ERROR_KEY)) {
+                                // If there was a client side error, check whether it was due to not enough data being available
+                                if (json.getString(CC_REST_API_ERROR_KEY).contains("NotEnoughValidWindowsException")) {
+                                    ccResponse.setNotEnoughDataForProposal(true);
+                                    result.complete(ccResponse);
+                                } else {
+                                    // If there was any other kind of error propagate this to the operator
+                                    result.fail(json.getString(CC_REST_API_ERROR_KEY));
+                                }
+                            } else {
+                                result.complete(ccResponse);
+                            }
+                        });
+                    } else {
+                        result.fail(new CruiseControlRestException(
+                                "Unexpected status code " + response.statusCode() + " for POST request to " +
+                                host + ":" + port + path));
```

Comment:
I think you should include the `statusMessage`. I also see scope here for some common error handling with the Kafka Connect client, but not for this PR. This comment applies to other similar places in this class.